permissions/permissions.go

```diff
@@ -0,0 +1,394 @@
+			permissionScopes(subusersStatsSums, read),
+			permissionScopes(subusersSummary, read),
+		)
+	}
+}
+
+func suppressionsPermissions() permissionsAccessor {
+	return func() []string {
+		return merge(
+			permissionScopes(Suppression, create, read, update, delete),
+			permissionScopes(suppressionBounces, create, read, update, delete),
+			permissionScopes(suppressionBlocks, create, read, update, delete),
+			permissionScopes(suppressionInvalidEmails, create, read, update, delete),
+			permissionScopes(suppressionSpamReports, create, read, update, delete),
+			permissionScopes(suppressionUnsubscribes, create, read, update, delete),
+		)
+	}
+}
+
+func teammatesPermissions() permissionsAccessor {
+	return func() []string {
+		return permissionScopes(Teammates, create, read, update, delete)
+	}
+}
+func templatesPermissions() permissionsAccessor {
+	return func() []string {
+		return merge(
+			permissionScopes(Templates, create, read, update, delete),
+			permissionScopes(templatesVersions, create, read, update, delete),
+			permissionScopes(templatesVersionsActivate, create, read, update, delete),
+		)
+	}
+}
+
+func trackingPermissions() permissionsAccessor {
+	return func() []string {
+		return merge(
+			permissionScopes(TrackingSettings, read),
+			permissionScopes(trackingSettingsClick, read, update),
+			permissionScopes(trackingSettingsGoogle, read, update),
+			permissionScopes(trackingSettingsOpen, read, update),
+			permissionScopes(trackingSettingsSubscription, read, update),
+		)
+	}
+}
+
+func userSettingsPermissions() permissionsAccessor {
+	return func() []string {
+		return merge(
+			permissionScopes(userAccount, read),
+			permissionScopes(userCredits, read),
+			permissionScopes(userEmail, create, read, update, delete),
+			permissionScopes(userMultifactorAuth, create, read, update, delete),
+			permissionScopes(userPassword, read, update),
+			permissionScopes(userProfile, read, update),
+			permissionScopes(userSettingsEnforcedTLS, read, update),
+			permissionScopes(userTimezone, read, update),
+			permissionScopes(userUsername, read, update),
+		)
+	}
+}
+
+func webhookPermissions() permissionsAccessor {
+	return func() []string {
+		return merge(
+			permissionScopes(userWebhooksEventSettings, read, update),
+			permissionScopes(userWebhooksEventTest, create, read, update),
+			permissionScopes(userWebhooksParseSettings, create, read, update, delete),
+			permissionScopes(userWebhooksParseStats, read),
+		)
+	}
+}
+
+func whitelabelPermissions() permissionsAccessor {
+	return func() []string {
+		return merge(
+			permissionScopes(Whitelabel, create, read, update, delete),
+		)
+	}
+}
+
+func whitelistPermissions() permissionsAccessor {
+	return func() []string {
+		return merge(
+			permissionScopes(accessSettingsWhitelist, create, read, update, delete),
+			permissionScopes(accessSettingsActivity, read),
+		)
+	}
+}
+
+func GetPermissions(permissionClass string, accessType string) []string {
```

Comment:
maybe instead of checking if `accessType` matches the global `fullAccess` variable we could have a `fullAccess` bool parameter. 
Maybe instead of a `accessType` string param we should have a `fullAccess` param. And we can remove the `fullAccess` global variable.
Then you could just check if `fullAccess` is true:
```golang
	if fullAccess {
 		return permissionScopes
 	}
```

Reply:
SUre